cmake/Utils.cmake

```diff
@@ -112,13 +112,11 @@ function(xgboost_set_cuda_flags target)
   if(USE_DEVICE_DEBUG)
     target_compile_options(${target} PRIVATE
       $<$<AND:$<CONFIG:DEBUG>,$<COMPILE_LANGUAGE:CUDA>>:-G;-src-in-ptx>)
-  else()
-    target_compile_options(${target} PRIVATE
-      $<$<COMPILE_LANGUAGE:CUDA>:-lineinfo>)
```

Comment:
Can we create a new CMake option to enable the lineinfo? Or is it redundant given the `USE_DEVICE_DEBUG` option?